kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesAttributesExtractor.java

```diff
@@ -80,6 +84,7 @@
   private static final String SCHEME = "http";
   private static final String HOST = "localhost";
   private Map<String, CustomResourceDefinitionContext> crdContexts;
+  private Map<String, String> pluralToKind = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
We might get some collisions here.
It's very unlikely but 2 different resources might have the same plural that translates into 2 different kinds. The situation is even more improbable since custom resources are handled separately, and I don't think any official resource or supported extension might be affected.
Anyway, just pointing this out.

Reply:
Let me refine things to also address #3216 and I'll remove any chance from ambiguity from built-in and custom types.